### PR TITLE
Code Review :Math magicians: components

### PR DESCRIPTION
**Pull request by amiraabouhadid**
## Description

A single-page app that runs in the browser with no errors was setup using create react app (CRA)

### General requirements

- [x] There are no linter errors.
- [x] Followed the **Github flow**.
- [x] The documentation is professional.

### HTML/CSS/JS requirements

- [x] We followed the list of [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).
- [x] We followed the list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

### Project requirements

- Delete all the boilerplate from CRA (text, images, styles).
- Create a directory called components.
- Inside components, create a new Calculator.js file.
- In Calculator.js, create a React component that matches the given design.
- Don't add any extra design.
- Develop just the UI of the calculator; the math logic to make it work will be added in the next project.
- Use class-based components.
- Once the Calculator is ready, import it and use it in your main component (App.js).
- Test it in the browser; it should render without problems.

Thank you for taking the time to review this pull request.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,4 +1,6 @@
-import './App.css';
+import Calculator from './components/Calculator';
 
-function App() {}
+function App() {
+  return <Calculator />;
+}
```
Comment (rustam-tolipov) [inline src/App.js]: _Your project works perfectly! 👏_

 However there is a small thing which need to be changed before approving it. 
- [x] Please turn `App.js` component into `class-based` component.

File: src/components/Calculator.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
+/* eslint-disable react/jsx-key */
+/* eslint-disable no-undef */
+/* eslint-disable no-unused-expressions */
+/* eslint-disable-next-line react/jsx-key */
```
Comment (Rustamxon7) [inline src/components/Calculator.js:4]: 👍 You did a great job by commenting your code in order to avoid pure-function linter errors!

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hi @amiraabouhadid , 

Good job so far!

emojibest_com_AnimatedSticker


### HIGHLIGHTS!

- No linters errors.💯
- Descriptive Readme.md. 💯
- Awesome UI with additional features

There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️

_Also check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js (hunk 1 above)
Comment (amiraabouhadid) [reply src/App.js]: Hello @Rustamxon7 , 

Thanks for your feedback. It is not a requirement to record a video for this project so I am confused about what you said regarding the microphone not working? I haven't attached a video as it was not required.

Comment (rustam-tolipov) [reply src/App.js]: > Hello @Rustamxon7,
> 
> Thanks for your feedback. It is not a requirement to record a video for this project so I am confused about what you said regarding the microphone not working? I haven't attached a video as it was not required.

Oh, sorry about that it's my bad! Yes, there is not that kind of thing to do!

Comment (amiraabouhadid) [reply src/App.js]: Moreover, as App.js does not need a state, isn't it better to use a stateless function?  As when I use a class-based component, I get a linter error saying: Component should be written as a pure function react/prefer-stateless-function. 
Please advise on why App.js needs to be a class-based component at this stage when only the design is required.

Comment (rustam-tolipov) [reply src/App.js]: In this case, it is the requirement to do more practice on class-based components.

**Review by elyor-doniyorov — APPROVED** [review]
Hi @amiraabouhadid, 

In my opinion, your project milestone is ready!  There is nothing else to say other than... well done! :shipit: 
- [OPTIONAL] All the boilerplate that got created by the CRA (texts, images, and styles) ought to have gotten deleted.

Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**